### DISCUSSION TITLE
Mise à jour du template des fiches (cycles de vie, niveau de complexité, ...)

### DISCUSSION THREAD

**Discussion by florinesueur** [cnumr/best-practices-wordpress#39] [General] [answered]
Nous devons nous mettre d'accord sur le template des fiches. J'ai essayé de faire un synthèse de la réunion et des discussions.

1. Utilisation de l'indicatif

2. Titre avec verbe (ex : Mobile first ou Concever Mobile First)

3. Utiliser plutôt le nous que le vous

4. Proposition du cycle de vie, 1 cycle par fiche (ce qui aidera aussi à la numérotation) :
- Analyse des besoins
- Installation
- Conception & Design 
- Intégration & Développement 
- Contenu & médias (Faire vivre son site) 
- Tests & validation 
- Mise en ligne 
- Maintenance 
- Fin de vie 
Document de mon expérience à l'installation d'un wordpress qui explique le pourquoi du cycle : https://1drv.ms/w/s!Ar8bO6aomsOlgY1GeCc2Z0LrPFB5og?e=B5O2Cu

5. Structurer la fiche en fonction des personas code/no-code. Comment l'écrire ? Si vous êtes design ou développeur....

6. Mettre un niveau de complexité et d'emprunte environnementale pour chaque persona

7. Utiliser [[]] quand il est nécessaire d'avoir une définition dans le lexique (qui pourra se faire en dernier)

8. Utiliser ==texte== pour mettre en highlight au besoin

9. Respecter la structure
| Le nombre ... | est inférieur ou égal à |
| ------------- | :---------------------: |
| de ici mettre la variable       |            0            |

**Comment by dNicolle** [comment]
Super ce récap,

1. / 3.
Plutôt indicatif ou Nous dans les fiches finalement ? Pour ma part, les fiches très didactic seront plus difficile à passer en nous, contrairement aux fiches plus rédactionnelles

2. ok pour revoir tous les titres de fiches avec un verbe

4. Je suis OK pour ce cycle de vie

5. La méta n'est pas suffisante pour toi @florinesueur ?

6. OK pour moi

7./8. OK

9. il faudra vérifier cette nomenclature sur chaque fiche oui

**Reply by florinesueur** [reply]
La meta lyfecycle est suffisante pour moi.
Savoir si cette meta accepte une phase du cycle ou plusieurs ?

**Reply by hrenaud** [reply]
À l'heure actuelle, la génération du site ne prend pas d'autre format de méta que ceux en place. Faire un changement implique de le répercuter dans le code Gatsby, mais il ne faut pas que cela nous freine.

Actuellement voici le format des métas :

```
title: 02. Limiter l'autocomplétion lors d'une recherche
people: Renaud H.
scope: Front-office
state_validation: TO Validate
responsible:
  - Code(use·ur) → Développeu·se·r
  - Designeu(se·r)
  - Low-code → Freelance et développeur Front-End d'agences
  - No-code → Madame et Monsieur tout le monde
lifecycle:
  - 2 - Conception / Design
priority_implementation: Fort 👍👍👍
environmental_impact: Fort 🌱🌱🌱
saved_resources:
  - ALL
path: /fiches/02-limiter-l-autocompletion-lors-d-une-recherche
```
On avait supprimé la complexité suite à nos échanges précédents.

Si nous voulons refléter nos échanges d'hier, il faudrait aller vers ça je pense :

```
title: 02. Limiter l'autocomplétion lors d'une recherche
people: Renaud H.
scope: Front-office
state_validation: TO Validate
priority_implementation: Fort 👍👍👍
saved_resources:
  - ALL
responsible:
  - Code(use·ur) → Développeu·se·r
  - Designeu(se·r)
  - Low-code → Freelance et développeur Front-End d'agences
  - No-code → Madame et Monsieur tout le monde
complexity:
  - Code(use·ur) → Développeu·se·r | Env Impact → Bas 🌱
  - Designeu(se·r) | Env Impact → Bas 🌱
  - Low-code → Freelance et développeur Front-End d'agences | Env Impact → Moyen 🌱🌱
  - No-code → Madame et Monsieur tout le monde | Env Impact → Fort 🌱🌱🌱
lifecycle:
  - 2 - Conception / Design
path: /fiches/02-limiter-l-autocompletion-lors-d-une-recherche
toIndex: true
```

- J'ai ajouté `toIndex` pour indiquer à Gatsby de générer ou non des pages/fiches (je l'ai mis en place sur le Lexique à cause du Sommaire). Il faudra le faire sur les persona.
- J'ai donc changé l'arbo en remettant `complexity` . J' ai aussi cumulé le `persona` à `environmental_impact`, que j'ai supprimé en tant que méta isolée.

Avec une fiche qui à **toutes** les persona, c'est chiant, il faut l'avouer, mais je ne vois pas comment faire autrement :)

> Avoir `responsible` et `complexity` semble redondant, mais je me sert de responsible pour faire un lien vers la persona citée. Si on laisse tomber ce lien, on peut supprimer `responsible`.

**Reply by hrenaud** [reply]
On avait aussi dit qu'il fallait sortir le code Gatsby de la branch `main`.

Je vais voir comment faire.

Par contre, il ne faudra pas faire de changement d'arborescence pour toutes les `best-practices-WordPress`/`personnas`/`lexique`, il faudra conserver le dossier `.github` à la racine et il ne faudra pas supprimer ces deux branches (que je ne peux pas protéger) :
- `site-gatsby-generator`
- `gh-pages`

**Reply by hrenaud** [reply]
Il faudrait réécrire les **persona** d'ailleurs, car elles sont encore à l'état brut, datant de nos toutes premières discussions.

**Comment by Yanngautreau** [comment]
N'ayant pu être malheureusement pas présent lors de la réunion, permettez moi de vous dire que je suis entièrement d'accord avec les modifs proposés notamment sur les no-codes (cela me ravi même pensant avant que le guide ne serait pas facilement abordable pour les non codeurs), cela va permettre de proposer du contenu, et de proposer des tips pour les no-codes pour qu'il tende vers l'éco-conception, Et un grand oui au cycle de vie. Bref, au top.

**Comment by dNicolle** [comment]
Je me suis trompée dans les numéros des points dans ma première réponse :/

5. Structurer la fiche en fonction des personas code/no-code. Comment l'écrire ? Si vous êtes design ou développeur....
-> c'est là que ma question devait être : "Les métas ne suffisent pas selon toi @florinesueur ,?"

6. Mettre un niveau de complexité et d'emprunte environnementale pour chaque persona
> Ok pour moi

@hrenaud 
j'ai dû mal à comprendre la partie Complexity
 - le fait de nommer le personna donne la complexité que cela représente, et par défaut l'impact économisé ?

Que fait-on des personnas ? on garde les 4 ? ou on réduit à 3 comme proposé par @florinesueur ou que 2 pour simplifier encore ?

**Comment by florinesueur** [comment]
Bonjour,

Je vois qu'il y a peut-être une incompréhension sur le 5 et 6. Du coup, je vais faire sur un template pour que vous voyez ce que je proposais.

---
title: <NUM de la fiche sur deux digits>. <NOM de la fiche>
path: /fiches/<NUM de la fiche sur deux digits>-<NOM de la fiche>
people: <Noms des rédacteurs>
scope: <Famille de la fiche>
state_validation: <Status le création de la fiche>
responsible: <Liste des cible(s) visé(s), Personnas concernée(s)> => Ne pas afficher sur le site
lifecycle: <Phase de "conception" concernée> => **Un cycle par fiche** 
priority_implementation: <Priorité de mise en œuvre> => **Une priorité par fiche**
environmental_impact: <Niveau de l'impact écologique> => **Impact le plus important des solutions proposées**
saved_resources: <liste des ressources économisées>
toIndex: <boolean: true|false>
---

## Sujet

Description de l'information du la fiche

## GreenIT vous conseille

Indication GreenIT général

### Solution no-code 🌱

Si solution pour les novices

### Solution code pas à pas 🌱🌱🌱

Si solution pour les novices avec accompagnement comme un tuto (1. Télécharger un FTP 2. etc)

### Solution code 🌱🌱🌱

Si solution code existante

## Exemple (les exemples peuvent inscrit directement dans les différents blocs de solution, si existant)

## Principe de validation

| Le nombre ... | est inférieur ou égal à |
| ------------- | :---------------------: |
| de [Indiquer ici les éléments] ...        |            0            |

## Sources
- Source s'il y en a

**Reply by florinesueur** [reply]
Du coup, j'avais compris que Fred proposait des paragraphes plutôt avec des titres comme :
- Si vous êtes designer....
- Si vous êtes développeur...

Concraitement ce que j'ai contre les persona en visible pour l'utilisateur, c'est que ça range les gens dans des cases et que je préfère ranger les solutions dans des cases.

**Reply by dNicolle** [reply]
Je suis OK avec ce template sans nommer de personnas, mais plutôt en indiquant les types de solutions.
Du coup, est-ce que le "responsible" serait toujours d'actualité ? (ou alors pour le site permettre de faire des tris par type de solutions proposées?)

**Reply by florinesueur** [reply]
Arffff, j'ai perdu mon comment :(

Pour les metas, je ne comprends plus trop. Du coup, je propose une simplification; il y en a 3 :
1. 🌱environmental_impact (notation sur 3) : à indiquer à chaque paragraphe de solution (si présent) + mettre le plus élevé en meta pour la card
2. 🧠 (peut-être qu'il y a un autre emoticon) complexity (notation sur 3) : Est ce necessaire de le mettre ? Pour moi non car nous essayons d'apporter justement des solutions de différents niveaux (no code, code tuto et code) pour les personas. Si oui, uniquement sur les différents blocs
3. 👍priority_implementation (notation sur 3) : Doit-il vraiment varier en fonction de la solution ? Est ce qu'elle ne peut pas être globale ? Si non, à indiquer à chaque paragraphe de solution (si présent) + mettre le plus élevé en meta pour la card

**Reply by dNicolle** [reply]
De mon point de vue : 
 - 🌱environmental_impact : Ok avec la proposition de @florinesueur :à indiquer à chaque paragraphe de solution (si présent) + mettre le plus élevé en meta pour la card
  - complexity : ne pas mettre puisque dépend de la solution que l'utilisateur va choisir
   - 👍priority_implementation : dépend du thème et non de la solution : donc par fiche

**Reply by florinesueur** [reply]
J'ai modifié le template (commentaire au dessus) dans ce sens. Est ce plus clair pour l'utilisateur ?

**Reply by florinesueur** [reply]
> Du coup, est-ce que le "responsible" serait toujours d'actualité ? (ou alors pour le site permettre de faire des tris par type de solutions proposées?)

Personellement, je ne l'afficherai pas sur le site. Est ce qu'un filtre `responsible` sera utilisé ? Je pense que dans un premier temps, ne le faisons pas puis s'il y a des demandes nous feront le filtre puisque de toute façon les metas sont déjà faites.

**Reply by hrenaud** [reply]
- On peut masquer la notion de persona sur tout le site, ça ne pose pas de problème. Je supprime les liens et les pages liste/personnas.
- On peut masquer `responsible` puisque la notion de persona disparait.

Le fait d'afficher 
- dans le corps du texte type de solution + impact, c'est top.
- dans la meta `environmental_impact`, le plus bas impact. C'est top aussi (il est visible sur la liste des fiches, pas que dans la fiche).

Si l'on souhaite conserver `complexity` en tant que meta (même invisible), il faudra trouver une solution pour la cumuler à la solution.
Si on pense cette propriété comme un Array, on peut mettre ce que l'on veut dedans comme libellé (ex. Nom de la solution + `environmental_impact`). On reporte à l'identique ce qui est dans le corps de page.

---

Fait-on aussi une page avec la liste des contributeurs au projet ?

**Reply by florinesueur** [reply]
Plus j'y réflechis et plus je trouve que `complexity` nous embête plus qu'autre chose. Est ce vraiment un apport d'avoir cette donnée subjective ? Sinon comment la rendre objective ? Si vous êtes ok, on ne s'en occupe pas en v1 et on voit ça en v2, si demande et si nécessaire ?

Je vais revenir vers vous avec un index du site qui reprendra une partie de la préface + d'autres éléments, notamment les contributeurs. Si c'est ok ?

**Reply by dNicolle** [reply]
OK pour les deux points !

**Reply by hrenaud** [reply]
Je suis sur la fiche **20. Concevez vos sites en Mobile First**.

Pour moi, clairement, elle s'adresse principalement aux Designers d'interface, donc il va falloir garder cette personna. Je réduirai à ça.

  - Code(use·ur)/Low-Code(use·ur)
  - Designeu(se·r) → UX/UI
  - No-code → Madame et Monsieur tout le monde

> Qu'on masque sur le site, mais qu'on conserve sur les fiches, non ?

---

Est-ce qu'on peut statuer et fermer ce sujet ? Qui pose dans le marbre les décisions sur les métas, titres de sections, etc. ?

Après, il faudra les reporter dans les fiches et dans les docs/readme.

**Reply by florinesueur** [reply]
Qu'on masque sur le site, mais qu'on conserve sur les fiches, non ? Oui et nous aviserons sur ce point en v2 en fonction des retours, besoins,..

Le plus simple est de voter. Nous sommes 5 acfis donc 3 pouces en l'air c'est adopté / 3 pouces en bas c'est refusé.
Si j'en oublie, n'hésitez pas à les rajouter.

**Comment by florinesueur** [comment]
Mettre un pouce en l'air si accepté et un pouce en bas si réfusé. @dNicolle @hrenaud @Yanngautreau @yannkozon 

Adoption du format pour les titres (répercuter sur le nom du .md et le path) : Verbe à l'infinitif + indication de la bonne pratique
Ex : Utiliser des polices de caractères standard

**Comment by florinesueur** [comment]
Mettre un pouce en l'air si accepté et un pouce en bas si réfusé. @dNicolle @hrenaud @Yanngautreau @yannkozon 

Adoption du format du cycle suivant :

- Analyse des besoins
- Installation
- Conception & Design
- Intégration & Développement
- Contenu & médias (Faire vivre son site)
- Tests & validation
- Mise en ligne
- Maintenance
- Fin de vie

**Comment by florinesueur** [comment]
Mettre un pouce en l'air si accepté et un pouce en bas si réfusé. @dNicolle @hrenaud @Yanngautreau @yannkozon 

Adoption d'attribuer une étape de cycle par BP

**Comment by florinesueur** [comment]
Mettre un pouce en l'air si accepté et un pouce en bas si réfusé. @dNicolle @hrenaud @Yanngautreau @yannkozon 

Adoption de masquer la notion de `persona` sur tout le site

**Reply by hrenaud** [reply]
J'ai mis un 👍🏻 Mais il faut en parler dans le préambule et trouver un moyen d'en parler sur le site pour que les DEV n'ai pas l'impression qu'ils n'aient rien à apprendre et à l'inverse que **Madame, Monsieur, tout le monde** ou les **Designers**, s'ils voient du code ne se disent pas que tout ça n'est pas pour eux...

**Reply by florinesueur** [reply]
Tout à fait ! Très bonne remarque. Cette notion devra apparaître sur l'index du site.

**Comment by florinesueur** [comment]
Mettre un pouce en l'air si accepté et un pouce en bas si réfusé. @dNicolle @hrenaud @Yanngautreau @yannkozon 

Adoption de laisser la notion de `persona` sur toute les fiches et les nouvelles en création

**Comment by florinesueur** [comment]
Mettre un pouce en l'air si accepté et un pouce en bas si réfusé. @dNicolle @hrenaud @Yanngautreau @yannkozon

Adoption de structurer, si possible, les fiches avec les paragraphes suivant :
- Solution no code
- Solution code pas à pas
- Solution code

**Reply by hrenaud** [reply]
Et, si aucune solution n'existe pour l'un des sujets, on peut ne rien proposer ?

**Reply by florinesueur** [reply]
Si par exemple c'est un point de vigilence ou d'avertissement, il n'est pas nécessaire d'utiliser ces paragraphes. C'est si on a besoin et si on recommande différents niveaux de solution.

**Comment by florinesueur** [comment]
Mettre un pouce en l'air si accepté et un pouce en bas si réfusé. @dNicolle @hrenaud @Yanngautreau @yannkozon

Adoption que `priority_implementation` soit attribuer à la BP (et non à la solution, si proposition de plusieurs no code, code p à p ou code)

**Comment by florinesueur** [comment]
Mettre un pouce en l'air si accepté et un pouce en bas si réfusé. @dNicolle @hrenaud @Yanngautreau @yannkozon

Adoption que `environmental_impact` soit attribuer à la BP si une seule proposition de solution. Si divers solutions proposées (no code / code), une valeur est attribuée à chaque paragraphe et la valeur de la BP reprend la valeur de la solution la plus grande.

**Comment by florinesueur** [comment]
Mettre un pouce en l'air si accepté et un pouce en bas si réfusé. @dNicolle @hrenaud @Yanngautreau @yannkozon

Adoption de l'abandon de `complexity`.

**Comment by florinesueur** [comment]
Mettre un pouce en l'air si accepté et un pouce en bas si réfusé. @dNicolle @hrenaud @Yanngautreau @yannkozon

Adoption pour la simplification de la liste `persona` :
- Code(use·ur)/Low-Code(use·ur)
- Designeu(se·r) → UX/UI
- No-code → Madame et Monsieur tout le monde

**Comment by florinesueur** [comment, marked as answer]
Readme et Template mis à jour.
